Realm/RLMSyncSession.h

```diff
@@ -150,8 +152,20 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (nullable RLMProgressNotificationToken *)addProgressNotificationForDirection:(RLMSyncProgressDirection)direction
                                                                           mode:(RLMSyncProgress)mode
+                                                           dispatchToMainQueue:(BOOL)dispatchToMainQueue
```

Comment:
But after the rest of this PR it's _not_ run on the sync client thread, it's run on a separate notification queue. The only thing they'll delay by doing expensive work is further progress callbacks.